experiment_helpers/check_rmsThresh.m

```diff
@@ -1,30 +1,119 @@
+%       * rmsThresh. If the RMS value is below rmsThresh, the output
+%           parameter bGoodTrial will be 0. This parameter is overridden by
+%           the 2nd input param `rmsThresh`, if that input param is used.
+%
+% 
 
 if nargin < 3, subAxis = []; end
+if nargin < 4
+    params = struct;
+end
+defaultParams.checkMethod = 'mean';
+defaultParams.limits = [0.037, 0.100; 0 0];
+defaultParams.rmsThresh = 0.037;
+params = set_missingFields(params, defaultParams, 0);
 
-max_rms = max(data.rms(:,1));
-if max_rms > rmsThresh %&& length(find(ost == 2)) > (offset(2) + 1)
-    bGoodTrial = 1;
-else
-    bGoodTrial = 0;
+% for backwards compatibility, the second input parameter can override params.rmsThresh
+if nargin >= 2 && ~isempty(rmsThresh)
+    params.rmsThresh = rmsThresh;
 end
 
 if isgraphics(subAxis)
+    %% plot Good and Warn shaded regions
     subplot(subAxis)
     tAxis = 0 : data.params.frameLen : data.params.frameLen * (size(data.fmts, 1) - 1);
     yyaxis left
     plot(tAxis/data.params.sr,data.rms(:,1));
     ylim([0 0.1])
     xlim([0 tAxis(end)/data.params.sr])
-    hline(rmsThresh,'k',':');
+
+    yGood = [params.limits(1,1) params.limits(1,1) params.limits(1,2) params.limits(1,2)];
+    yWarn = [params.limits(2,1) params.limits(2,1) params.limits(2,2) params.limits(2,2)];
+
+    colorWarn = [1,   1,   0.3];
+    colorGood = [0.4, 1,   0.4];
+
+    xPatchShade = [0, length(data.rms), length(data.rms), 0];
+
+    patch(xPatchShade,yWarn,colorWarn, 'FaceAlpha', 0.3, 'EdgeColor', 'none')
+    patch(xPatchShade,yGood,colorGood, 'FaceAlpha', 0.3, 'EdgeColor', 'none')
+
+    %% If evaluating based on peak amplitude, do that
+    switch params.checkMethod
+        case 'peak'
+            rmsValue = max(data.rms(:,1));
+        case 'mean'
+            %% determine onset and offset
+            % if OST onset and offset exist, use that
+            if any(data.ost_stat == 4)
+                % Finding the last instance of status 0 implies that the next status
+                % was 1, and did eventually successfully become status 2.
+                onset = 1 + find(data.ost_stat == 0, 1, 'last');
+                offset = 1 + find(data.ost_stat == 2, 1, 'last');
+                rmsValue = mean(data.rms(onset:offset, 1));
+
+                % if no ost tracking, use RMS data to find onset/offset
+            elseif ~any(data.ost_stat >= 1) && any(data.rms(:, 1) > 0.03)
+                onset = find(data.rms > 0.01, 1, 'first') + 5;
+                offset = find(data.rms(:, 1)<0.03 & data.rms(:, 1)>0.02 & data.rms_slope<0, 1, 'first') - 5;
+
+                % use middle 80%
+                onset = floor(onset + ((offset-onset)/10));
+                offset = ceil(offset - ((offset-onset)/10));
+                rmsValue = mean(data.rms(onset:offset, 1));
+
+                % can't determine rms mean, because no ost-based vowel found, and RMS too low
+            else
+                rmsValue = NaN;
+            end
```

Comment:
This is OK for today but I'd like to rework this so the rms calculation is not happening inside this `isgraphics` if statement. Only the plotting should happen in here. I'd suggest moving lines 63-89 before the isgraphics if statement, and replacing line 65 with something like:
```
[rmsValue,onset] = max(data.rms(:,1));
offset = offset;
```
or alternatively
```
rmsValue = max(data.rms(:,1));
onset = NaN;
offset = NaN;
```
Then you can end the switch/case. Afterwards, inside the isgraphics if statement, you can call the same plotting lines regardless of which case we're in.